### ISSUE TITLE
Support for sourceset modules configuration in idea plugin

### ISSUE THREAD

**Issue by AlexanderBartash** [gradle/gradle#30999]
This is related to https://github.com/gradle/gradle/issues/9681
This issue is relevant again because there is no option to disable "Project Structure -> Add -> import module -> select file -> Gradle -> uncheck create separate modules per source set" in IDEA anymore.

# Expected Behavior
I want to be able to modify the main & test modules like below:
example:
```
idea {
    modules {
        "main" {
            ...
        }
        "test" {
            ...
        }
    }
}
```
# Current Behavior
But currently I can only modify the empty module which is used to aggregate main + test.
```
idea {
    module {
        ...
    }
}
```

**Comment by abstratt** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by donat** [comment]
Can you achieve the desired configuration by using and configuring the `idea-ext` plugin? https://github.com/JetBrains/gradle-idea-ext-plugin/wiki#ideArtifacts

**Comment by AlexanderBartash** [comment]
@donat No. In my case I wanted to modify module's dependencies, I do not see such API in that plugin.
